gomint-server/src/main/java/io/gomint/server/network/packet/PacketWorldChunk.java

```diff
@@ -34,15 +37,19 @@ public PacketWorldChunk() {
     public void serialize( PacketBuffer buffer, int protocolID ) {
         buffer.writeSignedVarInt( this.x );
         buffer.writeSignedVarInt( this.z );
+        buffer.writeUnsignedVarInt( this.subChunkCount );
+        buffer.writeBoolean( this.cached );
         buffer.writeUnsignedVarInt( this.dataLength );
-        buffer.writeBytes( this.data, 0, this.dataLength );
+        buffer.writeBytes( this.data );
+        // buffer.writeBytes( this.data, 0, this.dataLength );
```

Comment:
is there a reason on why the full buffer is written? it can happen that the buffer is larged than the data which needs to be written (due to object pooling or overallocation in subchunk rendering)